### PR TITLE
inital smi-conformance action draft

### PR DESCRIPTION
**Pull request by s1ntaxe770r**
Signed-off-by: s1ntaxe770 <cyberdev01@protonmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/smi-conformace.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
```
Comment (leecalcote) [inline .github/workflows/smi-conformace.yml:12]: Oh, neato.

**Review by leecalcote — APPROVED** [review]
Looking good so far. If we’re able to make this reusable enough, there might be one action to perform either SMI Conformance or a performance test

File: .github/workflows/smi-conformace.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      - uses: actions/checkout@v2
+      
+      - name: Install mesheryctl
+        run: |
+          /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+          eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+          brew tap layer5io/tap
+          brew install mesheryctl
+          
+      - name: copy default config
```
Comment (leecalcote) [inline .github/workflows/smi-conformace.yml:36]: I don’t think this is necessary. You should be able to run a `system start` with `-y` which will generate your first meshconfig

File: .github/workflows/smi-conformace.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      - name: verify install
+        run: mesheryctl version
+        
+      - name: install go
+        run: sudo apt install golang-go
+        
+      - name: install kind
+        run: GO111MODULE="on" go get sigs.k8s.io/kind@v0.10.0 && kind create cluster
+        
+      - name: export kubeconfig
```
Comment (leecalcote) [inline .github/workflows/smi-conformace.yml:52]: Is this for Meshery’s sake or kubectl’s?

File: .github/workflows/smi-conformace.yml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        run: export TOKEN="${{ secrets.AUTH_TOKEN }}"
+        
+      - name: pipe to a file
+        run: |
+          cat <<EOF > ~/auth.json
+          { "meshery-provider": "Meshery", "token": "$TOKEN" }
+          EOF
+       
+      - name:  configure meshery
+        run:  mesheryctl system config minikube -t ~/auth.json && mesheryctl system logs
```
Comment (leecalcote) [inline .github/workflows/smi-conformace.yml:88]: I don’t believe that this will work with kind.

Comment (s1ntaxe770r) [reply .github/workflows/smi-conformace.yml:88]: Yeah, it doesn't. using the kubectl command you gave

File: .github/workflows/smi-conformace.yml (hunk 2 above)
Comment (s1ntaxe770r) [reply .github/workflows/smi-conformace.yml:36]: wow i seem to have missed this ,  would make changes on the next push

File: .github/workflows/smi-conformace.yml (hunk 3 above)
Comment (s1ntaxe770r) [reply .github/workflows/smi-conformace.yml:52]: Kubectl's

**Review by leecalcote — APPROVED** [review]
This initial workflow looks good. `mesheryctl mesh` has its own issues. Time to work on creating an Action out of this, which might involve renaming the repo.

**Comment by s1ntaxe770r** [comment]
for context i'm currently working on fixing [this](https://github.com/layer5io/meshery/issues/2923) issue

**Review by leecalcote — APPROVED** [review]

**Comment by ramrodo** [comment]
@s1ntaxe770r is this still a WIP? If not, please change this Draft into a PR ready to review

File: .github/workflows/smi-conformace.yml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      - uses: actions/checkout@v2
+      
+      - name: Install mesheryctl
+        run: |
+          /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+          eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+          brew tap layer5io/tap
+          brew install mesheryctl
+          
+
```
Comment (ramrodo) [inline .github/workflows/smi-conformace.yml:36]: Two blank lines. Remove one.

File: .github/workflows/smi-conformace.yml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+             kubectl create namespace meshery
+             helm install meshery --namespace meshery install/kubernetes/helm/meshery 
+           
+      - name: deploy using mesheryctl
+        if: ${{ env.DEPLOYMENT_MODE == 'mesheryctl' }} 
+        run: |
+          mesheryctl system start -y
+          mesheryctl system context create -p kubernetes        
+      
+
```
Comment (ramrodo) [inline .github/workflows/smi-conformace.yml:71]: Two blank lines. Remove one.

File: .github/workflows/smi-conformace.yml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+          EOF
+         
+      - name:  configure meshery
+        run: kubectl config view --flatten
+
+      - name: perfom smi conformance tests
+        run: mesheryctl mesh validate -a meshery-istio -t ~/auth.json --spec smi 
+       
+             
+
```
Comment (ramrodo) [inline .github/workflows/smi-conformace.yml:88]: Three blank lines. Remove two.

**Review by ramrodo — COMMENTED** [review]
Only few comments.

**Pull request closed without merging**